From 2538982e1da075ea0fe59c0debd26af87259d0f0 Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Mon, 22 Apr 2024 08:12:35 +0300
Subject: [PATCH] [SW-0] gaudi2: skip 48MB sram bist for debug

In DC setups, the boot flow is getting stuck at
different points in preboot, to debug this issue
This patch skips running SRAM bist.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I55c1b8234001edddf36ca726a75a5adb947b829b
---

diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index cdde13f..2a10b51 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -473,6 +473,10 @@
 		regmap_writel(cold_rst_data.data, mmCOLD_RST_DATA);
 	}
 
+	/* This is temporary change for debugging sram access issue */
+	preboot_puts("skipping BIST for 48MB SRAM - debug\n");
+	return;
+
 	if (cold_rst_data.bist_skip_enable) {
 		preboot_puts("bist skipped for 48MB SRAM\n");
 		return;
